runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -2904,281 +2904,308 @@ J9::CodeGenerator::populateOSRBuffer()
-      int32_t counter = inlinedCallSize;
-      TR_InlinedSiteHastTableEntry *orderedInlinedSiteListTable;
-      if (inlinedCallSize > 0)
-         {
-         orderedInlinedSiteListTable= (TR_InlinedSiteHastTableEntry*)self()->comp()->trMemory()->allocateMemory(sizeof(TR_InlinedSiteHastTableEntry) * inlinedCallSize, heapAlloc);
-         memset(orderedInlinedSiteListTable, 0, sizeof(TR_InlinedSiteHastTableEntry)*inlinedCallSize);
+         if (ramMethod != cg->comp()->getCurrentMethod()->getPersistentIdentifier()) // && info->_reloKind != TR_ValidateArbitraryClass)
+            {
+            int32_t i;
+            for (i = 0; i < cg->comp()->getNumInlinedCallSites(); i++)
+               {
+               TR_InlinedCallSite &ics = cg->comp()->getInlinedCallSite(i);
+               TR_OpaqueMethodBlock *inlinedMethod = fej9->getInlinedCallSiteMethod(&ics);
+
+               traceMsg(cg->comp(), "\tinline site %d inlined method %p\n", i, inlinedMethod);
+               if (ramMethod == inlinedMethod)
+                  {
+                  traceMsg(cg->comp(), "\t\tmatch!\n");
+                  siteIndex = i;
+                  break;
+                  }
+               }
+
+            if (i >= (int32_t) cg->comp()->getNumInlinedCallSites())
+               {
+               // this assumption isn't associated with a method directly in the compilation
+               // so we can't use a constant pool approach to validate: transform into TR_ValidateArbitraryClass
+               // kind of overkill for TR_ValidateStaticField, but still correct
+               (*info)->_reloKind = TR_ValidateArbitraryClass;
+               siteIndex = -1;   // invalidate main compiled method
+               traceMsg(cg->comp(), "\ttransformed into TR_ValidateArbitraryClass\n");
+               }
+            }
+
+         traceMsg(cg->comp(), "Found inlined site %d\n", siteIndex);
+
+         TR_ASSERT(siteIndex < (int32_t) cg->comp()->getNumInlinedCallSites(), "did not find AOTClassInfo %p method in inlined site table", *info);
+
+         cg->addExternalRelocation(new (cg->trHeapMemory()) TR::ExternalRelocation(NULL,
+                                                                          (uint8_t *)(intptr_t)siteIndex,
+                                                                          (uint8_t *)(*info),
+                                                                          (*info)->_reloKind, cg),
+                                                                          __FILE__, __LINE__, NULL);
          }
-      else
-         orderedInlinedSiteListTable = NULL;
+      }
+   }
 
-      TR_InlinedSiteLinkedListEntry *entry = NULL;
+static void addSVMValidationRecords(TR::CodeGenerator *cg)
+   {
+   TR::SymbolValidationManager::SymbolValidationRecordList &validationRecords = cg->comp()->getSymbolValidationManager()->getValidationRecordList();
+   if (cg->comp()->getOption(TR_UseSymbolValidationManager))
```

Comment:
by your earlier logic, wouldn't this list be empty if not using SVM?

Reply:
Yeah, same idea. However, the reason why they're two different functions is because the traditional validation records are added after relo records but **before** the inlining table records, whereas the SVM records are added **after** the inlining table records.